### PR TITLE
fix: implicit accept header can be overridden by format query

### PR DESCRIPTION
**Pull request by SgtPooki**
## Title

fix: implicit accept header can be overridden by format query

## Description

Fixes https://github.com/ipfs/helia-verified-fetch/issues/35 & also allows mapping of 'raw' to 'dag-json'. I imagine users will get errors if they try to accept dag-json for non-json content, we can address that later?

## Notes & open questions

Basically, we were previously doing the following:

1. check accept header
2. check format query param
3. if accept header exists, use it, otherwise, use the mapped query param format

However, that was causing the bug in https://github.com/ipfs/helia-verified-fetch/issues/35 because the format query param was being ignored if the accept header was present (which it usually is for most clients), even if it wasn't [a valid IPFS gateway Accept header](https://specs.ipfs.tech/http-gateways/path-gateway/#accept-request-header)

Now, we do this:

1. check accept header
2. check format query param
4. if accept header is not a valid IPFS gateway accept header, and we have a mapping for the format query param, use the mapped query param format value.
3. otherwise use the accept header

more notes are in https://github.com/ipfs/helia-verified-fetch/issues/35

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by SgtPooki — COMMENTED** [review]
self review

File: packages/verified-fetch/test/verified-fetch.spec.ts [hunk 1]
```diff
@@ -710,4 +710,63 @@ describe('@helia/verifed-fetch', () => {
...
+      const obj = {
+        hello: 'world'
+      }
+      const c = dagCbor(helia)
+      const cid = await c.add(obj)
+
+      const resp = await verifiedFetch.fetch(`http://example.com/ipfs/${cid}?format=dag-json`, {
+        headers: {
+          // see https://github.com/ipfs/helia-verified-fetch/issues/35
+          // accept: '*/*'
```
Comment (achingbrain) [inline packages/verified-fetch/test/verified-fetch.spec.ts:743]: Is this supposed to be commented out?

File: packages/verified-fetch/src/utils/is-accept-explicit.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { FORMAT_TO_MIME_TYPE } from './select-output-type.js'
+import type { ParsedUrlStringResults } from './parse-url-string.js'
+
+export interface IsAcceptExplicitOptions {
+
+  query?: ParsedUrlStringResults['query']
+  headers: Headers
+}
```
Comment (achingbrain) [inline packages/verified-fetch/src/utils/is-accept-explicit.ts:8]:
```suggestion
export interface IsAcceptExplicitOptions {
  query?: ParsedUrlStringResults['query']
  headers: Headers
}
```

File: packages/verified-fetch/src/utils/get-resolved-accept-header.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { isExplicitAcceptHeader, isExplicitFormatQuery, isExplicitIpldAcceptRequest } from './is-accept-explicit.js'
+import { queryFormatToAcceptHeader } from './select-output-type.js'
+import type { ParsedUrlStringResults } from './parse-url-string.js'
+import type { ComponentLogger } from '@libp2p/interface'
+
+export interface ResolvedAcceptHeaderOptions {
+  query?: ParsedUrlStringResults['query']
+  headers?: RequestInit['headers']
+  logger?: ComponentLogger
```
Comment (achingbrain) [inline packages/verified-fetch/src/utils/get-resolved-accept-header.ts:9]: `helia.logger` will always be set so I don't think this field needs to be optional.

**Review by achingbrain — CHANGES_REQUESTED** [review]
LGTM, comments inline.

**Review by achingbrain — APPROVED** [review]
LGTM, comments inline.

**Pull request merged**